main/src/library/FilePath.flix

```diff
@@ -147,11 +147,311 @@ pub eff FilePath {
+                    k(())
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: UnsupportedOperationException => Err(IOError.Generic(ex.getMessage()))
+                }
+            }
+
+            def appendBytes(data, f, k) = {
+                try {
+                    region rc {
+                        let bytes = Vector.toArray(rc, data);
+                        let opt: OpenOption = checked_cast(StandardOpenOption.APPEND);
+                        Files.write(Paths.get(f), bytes, ...{opt});
+                        k(())
+                    }
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: UnsupportedOperationException => Err(IOError.Generic(ex.getMessage()))
+                }
+            }
+
+            def truncate(f, k) = {
+                try {
+                    region rc {
+                        let opt: OpenOption = checked_cast(StandardOpenOption.TRUNCATE_EXISTING);
+                        let arr: Array[Int8, rc] = Array#{} @ rc;
+                        Files.write(Paths.get(f), arr, ...{opt});
+                        k(())
+                    }
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: UnsupportedOperationException => Err(IOError.Generic(ex.getMessage()))
+                }
+            }
+
+            def mkDir(d, k) = {
+                try {
+                    Files.createDirectory(Paths.get(d), (...{}: Vector[FileAttribute]));
+                    k(())
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: UnsupportedOperationException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: FileAlreadyExistsException => Err(IOError.Generic(ex.getMessage()))
+                }
+            }
+
+            def mkDirs(d, k) = {
+                try {
+                    Files.createDirectories(Paths.get(d), (...{}: Vector[FileAttribute]));
+                    k(())
+                } catch {
+                    case ex: InvalidPathException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: IOException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: UnsupportedOperationException => Err(IOError.Generic(ex.getMessage()))
+                    case ex: FileAlreadyExistsException => Err(IOError.Generic(ex.getMessage()))
```

Comment:
This inherits from IOException, so it will never be reached. It has to be moved upwards. I suggest to always have IOException last.